support/tarball/build.sh

```diff
@@ -12,5 +12,25 @@ export NUGET_PACKAGES="$SCRIPT_ROOT/packages/"
 
 MSBUILD_ARGUMENTS=("/p:OfflineBuild=true" "/flp:v=detailed" "/clp:v=detailed")
 
+echo "Expanding BuildTools dependencies into packages directory..."
+# init-tools tries to copy from its script directory to Tools, which in this case is a copy to
+# itself. This is an error. To avoid the error, use a temp dir that we immediately delete.
+TEMP_TOOLS_DIR="$SCRIPT_ROOT/ToolsTemp"
+PREBUILT_PACKAGE_SOURCE="$SCRIPT_ROOT/prebuilt/nuget-packages"
+(
+    set -x
+
+    "$CLI_ROOT/dotnet" restore "$SCRIPT_ROOT/init-tools.msbuild" --no-cache --packages "$SCRIPT_ROOT/packages" --source "$PREBUILT_PACKAGE_SOURCE" || exit $?
```

Comment:
The `|| exit $?` pattern is what I came up with to make sure it exits on the first error. `set -e` would normally do this, but it doesn't apply when the command uses `||`. (I used `||` on the subshell to notify the user that it failed rather than just having the script exit without any info.) Better ideas welcome.